config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurer.java

```diff
@@ -319,12 +318,11 @@ private Saml2AuthenticationRequestFactory getAuthenticationRequestFactory(B http
 		if (resolver != null) {
 			return resolver;
 		}
-		if (version().startsWith("4")) {
-			return new OpenSaml4AuthenticationRequestFactory();
-		}
-		else {
-			return new OpenSamlAuthenticationRequestFactory();
+		if (ClassUtils.isPresent("org.opensaml.core.xml.persist.impl.PassthroughSourceStrategy",
+				Version.class.getClassLoader())) {
```

Comment:
It feels a little odd to be calling for `Version`'s classloader given that we aren't working with the `Version` class. I wonder if passing `null` here would be better.